handlers.go

```
@@ -1026,6 +968,40 @@ func (s *Server) handleAdd(c echo.Context, u *User) error {
 	})
 }
 
+func (s *Server) redirectIfLocalContentAddingDisabled(c echo.Context, u *User) error {
+	uep, err := s.getPreferredUploadEndpoints(u)
+	if err != nil {
+		log.Warnf("failed to get preferred upload endpoints: %s", err)
+	} else if len(uep) > 0 {
+		// propagate any query params
+		req, err := http.NewRequest("POST", fmt.Sprintf("%s/content/add", uep[rand.Intn(len(uep))]), c.Request().Body)
+		if err != nil {
+			return err
+		}
+		req.Header = c.Request().Header.Clone()
+		req.URL.RawQuery = c.Request().URL.Query().Encode()
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+
+		c.Response().WriteHeader(resp.StatusCode)
+
+		_, err = io.Copy(c.Response().Writer, resp.Body)
+		if err != nil {
+			log.Errorf("proxying content-add body errored: %s", err)
```

same here, we should return the error